mqtt/mqtt-edgehub/src/command.rs

```diff
@@ -0,0 +1,248 @@
+use std::time::Duration;
+
+use futures_util::{future::BoxFuture, StreamExt};
+use lazy_static::lazy_static;
+use regex::Regex;
+use tokio::net::TcpStream;
+use tracing::{error, info, warn};
+
+use mqtt3::{
+    proto, Client, Event, IoSource, ShutdownError, SubscriptionUpdateEvent, UpdateSubscriptionError,
+};
+use mqtt_broker::{BrokerHandle, Error, Message, SystemEvent};
+
+const TOPIC_FILTER: &str = "$edgehub/+/disconnect";
+const CLIENT_EXTRACTION_REGEX: &str = r"\$edgehub/(.*)/disconnect";
```

Comment:
client_id is a bit more complicated https://github.com/Azure/iotedge/blob/iiot/mqtt/mqtt-edgehub/src/topic/translation.rs#L66

Reply:
By more complicated, what do you mean? I see that it can be of "device/module" or "device" form but both will be captured by the existing regex?

Reply:
I think this should be enough. In topic translation we need to extract deviceId and moduleId from the clientId, here we just need to know the string to drop the connection.